libp2p/Host.cpp

```diff
@@ -284,6 +307,7 @@ void Host::startPeerSession(Public const& _id, RLP const& _hello,
             m_peers[_id] = peer;
         }
     }
+    peer->m_lastHandshakeFailure = NoFailure;
```

Comment:
It's `NoFailure` when `Peer` is created, so better move this to `if (itPeer != m_peers.end())` case.

Reply:
Ah good point